ui/src/app/cron-workflows/components/cron-workflow-details/cron-workflow-details.tsx

```diff
@@ -87,6 +87,99 @@ export const CronWorkflowDetails = ({match, location, history}: RouteComponentPr
+        } else {
+            document.location.href = link.url;
+        }
+    };
+
+    const getItems = () => {
+        const items = [
+            {
+                title: 'Submit',
+                iconClassName: 'fa fa-plus',
+                disabled: edited,
+                action: () =>
+                    services.workflows
+                        .submit('cronwf', name, namespace)
+                        .then(wf => navigation.goto(uiUrl(`workflows/${wf.metadata.namespace}/${wf.metadata.name}`)))
+                        .then(() => setError(null))
+                        .catch(setError)
+            },
+            {
+                title: 'Update',
+                iconClassName: 'fa fa-save',
+                disabled: !edited,
+                action: () => {
+                    // magic - we get the latest from the server and then apply the changes from the rendered version to this
+                    return services.cronWorkflows
+                        .get(name, namespace)
+                        .then(latest =>
+                            services.cronWorkflows.update(
+                                {
+                                    ...latest,
+                                    spec: cronWorkflow.spec,
+                                    metadata: {...cronWorkflow.metadata, resourceVersion: latest.metadata.resourceVersion}
+                                },
+                                cronWorkflow.metadata.name,
+                                cronWorkflow.metadata.namespace
+                            )
+                        )
+                        .then(setCronWorkflow)
+                        .then(() => notifications.show({content: 'Updated', type: NotificationType.Success}))
+                        .then(() => setError(null))
+                        .then(() => setEdited(false))
+                        .catch(setError);
+                }
+            },
+            suspendButton,
+            {
+                title: 'Delete',
+                iconClassName: 'fa fa-trash',
+                disabled: edited,
+                action: () => {
+                    popup.confirm('confirm', 'Are you sure you want to delete this cron workflow?').then(yes => {
+                        if (yes) {
+                            services.cronWorkflows
+                                .delete(name, namespace)
+                                .then(() => navigation.goto(uiUrl('cron-workflows/' + namespace)))
+                                .then(() => setError(null))
+                                .catch(setError);
+                        }
+                    });
+                }
+            },
+            {
+                title: 'Share',
+                iconClassName: 'fa fa-share-alt',
+                action: () => setSidePanel('share')
+            }
+        ];
+
+        if (cronWorkflow?.spec?.workflowSpec?.workflowTemplateRef) {
```

Comment:
I did not know about “?” until recently.

Reply:
Yeah, its amazing 😄